### PR TITLE
Bjorncs/http2 stream idle timeout

### PR DESCRIPTION
**Pull request by bjorncs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: container-core/src/main/java/com/yahoo/jdisc/http/server/jetty/HttpRequestDispatch.java [hunk 1]
```diff
@@ -119,6 +120,10 @@ public void dispatch() throws IOException {
                             error,
                             () -> "Network connection was unexpectedly terminated: " + parent.jettyRequest.getRequestURI());
                     parent.metricReporter.prematurelyClosed();
+                } else if (isErrorOfType(error, TimeoutException.class)) {
```
Comment (jonmv) [inline container-core/src/main/java/com/yahoo/jdisc/http/server/jetty/HttpRequestDispatch.java:123]: Just jetty that sende these here?

File: container-core/src/main/java/com/yahoo/jdisc/http/server/jetty/ServletResponseController.java [hunk 2]
```diff
@@ -75,6 +76,9 @@ private static int getStatusCode(Throwable t) {
             return HttpServletResponse.SC_NOT_FOUND;
         } else if (t instanceof RequestException) {
             return ((RequestException)t).getResponseStatus();
+        } else if (t instanceof TimeoutException) {
+            // E.g stream idle timeout for HTTP/2
+            return HttpServletResponse.SC_SERVICE_UNAVAILABLE;
```
Comment (jonmv) [inline container-core/src/main/java/com/yahoo/jdisc/http/server/jetty/ServletResponseController.java:81]: Same here

**Review by jonmv — APPROVED** [review]
LGTM provided you answer those wuestions 🙃

File: container-core/src/main/java/com/yahoo/jdisc/http/server/jetty/HttpRequestDispatch.java (hunk 1 above)
Comment (bjorncs) [reply container-core/src/main/java/com/yahoo/jdisc/http/server/jetty/HttpRequestDispatch.java:123]: Exceptions thrown by com.yahoo.jdisc.handler.RequestHandler will be treated equally with Jetty's exceptions

File: container-core/src/main/java/com/yahoo/jdisc/http/server/jetty/ServletResponseController.java (hunk 2 above)
Comment (bjorncs) [reply container-core/src/main/java/com/yahoo/jdisc/http/server/jetty/ServletResponseController.java:81]: Same answer as above :)

**Pull request merged**